tests/gnmi/helper.py

```diff
@@ -9,36 +9,26 @@
 GNMI_SERVER_START_WAIT_TIME = 15
 
 
-def gnmi_container(duthost):
-    global GNMI_CONTAINER_NAME
-    if len(GNMI_CONTAINER_NAME) == 0:
-        if get_image_type(duthost) == "public":
-            GNMI_CONTAINER_NAME = "telemetry"
-        else:
-            GNMI_CONTAINER_NAME = "gnmi"
-    return GNMI_CONTAINER_NAME
-
-
-def gnmi_program(duthost):
-    global GNMI_PROGRAM_NAME
-    if len(GNMI_PROGRAM_NAME) == 0:
-        if get_image_type(duthost) == "public":
-            GNMI_PROGRAM_NAME = "telemetry"
+@lru_cache(maxsize=None)
+class GNMIEnvironment(object):
+    def __init__(self, duthost):
+        self.duthost = duthost
+        self.use_gnmi_container = duthost.shell("docker ps | grep -w gnmi", module_ignore_errors=True)['rc'] == 0
```

Comment:
>use_gnmi_container [](http://example.com/codeflow?start=13&length=18)

Similar concern, do you intent to check running status of config?

Reply:
Will user docker image instead.